src/mygrad/nnet/activations/softmax.py

```diff
@@ -7,6 +7,7 @@
 
 def _softmax(x, kwargs):
     x = x - x.max(**kwargs)
+    x = x.astype(np.float)
```

Comment:
We should cast to float only if x is an int-array. Otherwise if the user inputs `float32`, this will cast to `float64`